### PR TITLE
Query Default Retention Policy using full name - InfluxDB

### PR DESCRIPTION
**Pull request by shanielh**
Related to #4347 

**edit** : tests are not passing, I'll fix them if @torkelo want this fix.

### CODE REVIEW

File: public/app/plugins/datasource/influxdb/datasource.ts [hunk 1]
```diff
@@ -47,6 +47,9 @@ export function InfluxDatasource(instanceSettings, $q, backendSrv, templateSrv)
     // replace templated variables
     allQueries = templateSrv.replace(allQueries, options.scopedVars);
 
+    // replace database name
+    allQueries = allQueries.replace(/\$database/g, this.database);
```
Comment (torkelo) [inline public/app/plugins/datasource/influxdb/datasource.ts]: why did you add this?

Comment (shanielh) [inline public/app/plugins/datasource/influxdb/datasource.ts]: Since the query doesn't have the database name on render.

Comment (torkelo) [inline public/app/plugins/datasource/influxdb/datasource.ts]: but it is added in the request, https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/influxdb/datasource.ts#L152

https://github.com/grafana/grafana/blob/master/pkg/api/dataproxy.go#L44

Comment (shanielh) [inline public/app/plugins/datasource/influxdb/datasource.ts]: I'll check the documentation of `InfluxDB` and see if I can delete it and fix the pr.

**Comment by shanielh** [comment]
@torkelo I've updated the code and the specs.

**Comment by bergquist** [comment]
Thanks for help us make grafana better.

But this would cause all current panels that send queries without retention policy to stop working. That's not acceptable. I would prefer another solution to this problem.
